### ISSUE TITLE
Add support for headers to DjangoParser

### ISSUE THREAD

**Issue by richshaw2015** [marshmallow-code/webargs#540]
```
NotImplementedError: Header parsing not supported by DjangoParser
```

**Comment by sirosen** [comment]
The decision not to support this was made 7 years ago (i.e. circa Django 1.5), and I think django didn't support this well.
I think we can add support today -- I'll try to find the time to do it and test properly.

For now, I think you could just go with the following snippet to get support
```python
from webargs.djangoparser import DjangoParser

class Parser(DjangoParser):
    def load_headers(self, req, schema):
        return req.META

djangoparser = Parser()
use_args = djangoparser.use_args
use_kwargs = djangoparser.use_kwarg
```

**Comment by sirosen** [comment]
I just noticed that this was closed incorrectly.
I'd like to fix this in 7.0. Reopening and adding to the milestone.
